src/model/UserDataSource.ts

```diff
@@ -75,6 +118,7 @@ export default class UserDataSource extends MongoDataSource<User> {
       if (username != null) {
         usernameInfo = {
           username,
+          canonicalName: username.replaceAll(nonAlphanumericRegex, ''),
```

Comment:
I'm also storing `canonicalName` in the db.  It's a sanitized version of `username`, ie without separating characters (dash, underscore and period).   This allows us to easily account for variations when checking for uniqueness.

### Example:

Existing user:
- username: cool-user23
- canonicalName: cooluser23

New user's username
- cool.user23 ❌
- c.ool.user-23 ❌ 
- cooluser2023  ✅



See UserDatasource.usernameExists()
 https://github.com/OpenBeta/openbeta-graphql/blob/dd967802eb0d80ae121a1a8be7a2e83660c11120/src/model/UserDataSource.ts#L37)

Reply:
This makes sense.  Is there an opentacos PR with the corresponding error message for when a user runs into this?

Reply:
Is the goal here promoting uniqueness and expressiveness of usernames? To me it seems like we'd be shrinking the space of possible usernames while having to deal with special chars, which is a lose-lose situation. But I don't feel strongly!

Reply:
> Is the goal here promoting uniqueness and expressiveness of usernames? 

Not really. Just to make it easier to check some edge cases where someone unintentionally (or intentionally) creates **john-doe** when there's already a **johndoe**.  Gmail also does this check.  'user.name' and 'username' belong to the same user.

> Is there an opentacos PR with the corresponding error message for when a user runs into this?

If someone tries the 'john.doe' scenario above, the error is 'Username already exists". 

https://github.com/OpenBeta/open-tacos/blob/5c3d953c6416867d08ae3fe2e7c0d5a6869d52e2/src/components/users/account/UsernameChangeForm.tsx#L111-L125